File: protocol-generator/dart/test/conformance_test.dart
import 'package:entity_core_protocol/src/conformance/harness.dart';
import 'package:test/test.dart';

/// The wire-conformance gate (the S2 codec gate): every vector in the v7.71
/// corpus must be byte-identical (encode) / correctly rejected (decode_reject).
/// A FAIL here means the CODE is wrong (S5 discipline) — never the corpus.
void main() {
  test('wire-conformance corpus is byte-identical (69/69)', () async {
    final result = await ConformanceHarness.run(ConformanceHarness.defaultFixture());
    if (result.failures.isNotEmpty) {
      // ignore: avoid_print
      print(result.failures.join('\n'));
    }
    // ignore: avoid_print
    print('== ECF conformance: ${result.pass}/${result.total} PASS, '
        '${result.fail} FAIL ==');
    expect(result.fail, 0, reason: result.failures.join('\n'));
    expect(result.pass, result.total);
    expect(result.total, 69, reason: 'expected 69 testable vectors in v7.71');
  });
}

File: protocol-generator/typescript/test/conformance.test.ts
import { test } from "node:test";
import assert from "node:assert/strict";
import { loadCorpusBytes } from "./corpus.js";
import { runConformance } from "./conformance-runner.js";

/**
 * The S2 gate (node:test form). The standalone `run-conformance.ts` prints the
 * category table; this asserts byte-identity to the cross-blessed corpus inside
 * the `node --test` run.
 */
test("ECF conformance corpus — every vector byte-identical to the cross-blessed fixture", () => {
  const report = runConformance(loadCorpusBytes());
  const failures = report.results.filter((r) => !r.pass).map((r) => `${r.id} [${r.kind}]: ${r.message}`);
  assert.deepEqual(failures, [], `\nfailing vectors:\n${failures.join("\n")}\n`);
  assert.ok(report.results.length >= 69, `expected ≥69 vectors, ran ${report.results.length}`);
});
